src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -322,10 +356,79 @@ private void UnloadDatabaseInternal(string databaseName,[CallerMemberName] strin
             }
         }
 
+        public bool ShouldDeleteShardedContext(TransactionOperationContext context, string dbName, RawDatabaseRecord rawRecord)
+        {
+            // TODO: RavenDB-13106
+            var directDelete = rawRecord.DeletionInProgress?.Any(x => x.Key.StartsWith(_serverStore.NodeTag) && x.Value != DeletionInProgressStatus.No);
+            if (directDelete == false)
+                return false;
+
+            context.CloseTransaction();
```

Comment:
Why? this is a really strange place to do this

Reply:
this is optimization like [this](https://github.com/ravendb/ravendb/blob/fd2173c4c165fbbdfee8d1e3417236bd09678d43/src/Raven.Server/Documents/DatabasesLandlord.cs#L318) we know that we are deleting and context won't be used anymore